javascript/attributes.js

```diff
@@ -62,6 +64,38 @@ export const extractElementAttributes = element => {
       }
     }
   }
+  return attrs
+}
+
+// Extracts dataset of an element and includes inherited data attributes from parents if requested.
+//
+export const extractElementDataset = element => {
+  let attrs = extractDataAttributes(element) || {}
+
+  if (element.attributes[defaultSchema.reflexInheritAttribute]) {
+    let parent = element.parentElement
+
+    while (parent) {
+      attrs = { ...extractDataAttributes(parent), ...attrs }
```

Comment:
```suggestion
      attrs = { ...parent.dataset, ...attrs }
```
We could use `parent.dataset` here (and on line 73) instead of relying on the `extractDataAttributes()` function. But we would then need to convert the snakeCase key-names back to dasherized keys on the backend. 

But if we would do that, we could get rid of the `extractDataAttributes()` function.

Reply:
I think it's OK to keep this computation/responsibility on the client.